hooks/useAuth/useAuth.tsx

```diff
@@ -89,91 +65,143 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
+    const body: {
+      user: { [key: string]: string | number };
+      student?: { nusnetId: string; matricNo: string; cohortYear: number };
+      mentor?: { cohortYear: number };
+      adviser?: { cohortYear: number };
+      facilitator?: { cohortYear: number };
+    } = {
+      user: { email },
     };
-    if (name) body.user = { ...body.user, name };
-    if (matricNo) body.user = { ...body.user, matricNo };
-    if (nusnetId) body.user = { ...body.user, nusnetId };
+
+    if (user && name) {
+      user.name = name;
+    }
+
+    switch (role) {
+      case "students":
+        body.student = {
+          nusnetId: nusnetId ?? "",
+          matricNo: matricNo ?? "",
+          cohortYear,
+        };
+        break;
+      case "mentors":
+        body.mentor = {
+          cohortYear,
+        };
+        break;
+      case "advisers":
+        body.adviser = {
+          cohortYear,
+        };
+        break;
+      case "facilitators":
+        body.facilitator = {
+          cohortYear,
+        };
+        break;
+      default:
+        break;
+    }
+
     /**
      * attempt to create new user in postgres
      */
     const apiServiceBuilder = new ApiServiceBuilder({
       method: HTTP_METHOD.POST,
-      endpoint: `/${role}`,
+      endpoint: `/users/create-${role.slice(0, role.length - 2)}`, // remove the last "s" in the role name
       body,
     });
     const apiService = apiServiceBuilder.build();
     const createUserResponse = await apiService();
 
     /**
-     * terminate sign up if postgres user creation fails and display error message
+     * throw error if postgres user creation failed
      */
     if (!createUserResponse.ok) {
       const errorMessage = await createUserResponse.text();
       throw new Error(errorMessage);
     }
 
-    /**
-     * attempt to create new user in firebase
-     */
-    await createUserWithEmailAndPassword(auth, email, password).catch(
-      async (err) => {
-        /**
-         * delete user in postgres if firebase user creation fails
-         */
-        const apiServiceBuilder = new ApiServiceBuilder({
-          method: HTTP_METHOD.DELETE,
-          endpoint: `/${email}`,
-        });
-        const apiService = apiServiceBuilder.build();
-        const deleteUserResponse = await apiService();
-
-        /**
-         * terminate sign up after deleting user in postgres and display error message
-         */
-        if (!deleteUserResponse.ok) {
-          const errorMessage = await deleteUserResponse.text();
-          throw new Error(errorMessage);
-        } else {
-          throw new Error(
-            err instanceof Error
-              ? err.message
-              : "Something went wrong while signing up"
-          );
-        }
-
-        throw err;
-      }
-    );
+    setIsLoading(false);
   };
 
   const signIn = async (email: string, password: string) => {
-    await signInWithEmailAndPassword(auth, email, password);
+    setIsLoading(true);
```

Comment:
i think this is okay but we need to watch if this causes issues
(eg. signing in sets loading to true so if the landing page has any authIsLoading checks, it would cause the whole page to change and flicker)